pkg/daemon/update.go

```diff
@@ -1228,6 +1228,11 @@ func (dn *Daemon) updateFiles(oldConfig, newConfig *mcfgv1.MachineConfig) error
 	if err := dn.writeUnits(newIgnConfig.Systemd.Units); err != nil {
 		return err
 	}
+	// In OKD /var/home/core/.ssh/authorized_keys is not created during ignition.  Create
```

Comment:
I may have missed conversation around this(if already happened). What is the reason OKD by default writes to /var/home/core/.ssh/authorized_keys.d/ignition which differs from default OCP behavior?

Reply:
IIUC the new Ignition does that, @LorbusChris or @bgilbert might know more

Reply:
@sinnykumari  ignition 2.x writes to there by default unless there is an override environment variable set... 

In ignition:  internal/exec/util/passwd.go in AuthorizeSSHKeys()

```                
                if distro.WriteAuthorizedKeysFragment() {
                        path, err = u.JoinPath(usr.HomeDir, ".ssh", "authorized_keys.d", "ignition")
                        if err == nil {
                                err = writeAuthKeysFile(usr, path, []byte(ks))
                        }
                } else {
                        path, err = u.JoinPath(usr.HomeDir, ".ssh", "authorized_keys")
                        if err == nil {
                                err = writeAuthKeysFile(usr, path, []byte(ks))
                        }
                }
```

and from distro:
```
        // writeAuthorizedKeysFragment indicates whether to write SSH keys
        // specified in the Ignition config as a fragment to
        // ".ssh/authorized_keys.d/ignition" ("true"), or to
        // ".ssh/authorized_keys" ("false").
        writeAuthorizedKeysFragment = "true"
```
so its always true

but it looks like it can be over-written with an environment variable:

```func WriteAuthorizedKeysFragment() bool {
	return bakedStringToBool(fromEnv("WRITE_AUTHORIZED_KEYS_FRAGMENT", writeAuthorizedKeysFragment))
}

func fromEnv(nameSuffix, defaultValue string) string {
	value := os.Getenv("IGNITION_" + nameSuffix)
	if value != "" {
		return value
	}
	return defaultValue
}
```

I'm not sure how RHCOS/OCP is doing this but I'm guessing the environment variable is set in RHCOS...

Reply:
The environment variable is for CI testing.  The variables in the distro package can be overridden at compile time, and the RHCOS package does so.

`writeAuthorizedKeysFragment = "true"` is the recommended default for an OS, allowing multiple tools to independently manage their own sets of SSH keys.  But on RHCOS there is only one set of keys, and it was decided that Ignition and the MCO would manage `~/.ssh/authorized_keys` directly.

Reply:
RHCOS is but FCOS is not doing this and is using the default from ignition and creating authorized_keys.d/ignition which is not what MCO is expecting and because of https://github.com/openshift/machine-config-operator/pull/2087, using authorized_keys.d/ignition is disabled so there is a need to create authorized_keys for logging in to the nodes...

Reply:
#2087 was implemented specifically to address the difference in the paths of Ignition-written SSH keys between FCOS and RHCOS, ensuring that Ignition won't write keys on FCOS that the MCO then fails to manage.  The MCO still manages `~/.ssh/authorized_keys` on FCOS nodes.

Reply:
Except that since the authorized_keys.d/ignition file is never moved/copied to authorized_keys, ssh into the server is disabled, until and if you can update 99_master_ssh and/or 99_worker_ssh.

And if there are issues with building the cluster, debugging becomes very difficult as you cannot ssh into the nodes...